src/parse.rs

```diff
@@ -31,27 +31,38 @@ use crate::scanners::*;
 use crate::strings::CowStr;
 use crate::tree::{Tree, TreeIndex, TreePointer};
 
+/// Tags for elements that can contain other elements.
 #[derive(Clone, Debug, PartialEq)]
 pub enum Tag<'a> {
-    // block-level tags
+    /// A paragraph of text and other inline elements.
     Paragraph,
-    Rule,
+    /// A horizontal ruler.
+    Rule, // FIXME: shouldn't this just be an event?
```

Comment:
I'm trying to remember the rationale, there might not be a good one. This does feel more event-like than tag-like.